src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,152 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.exception.DuplicateDataException;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class SortCommand extends Command {
+    private static String argument;
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" + "Sorts all persons by a chosen criterion\n" +
+            "\tExample: " + COMMAND_WORD + " name/phone/email";
+    public static final String MESSAGE_EMPTY = "Addressbook is empty";
+    public static final String MESSAGE_WRONG_ARGUMENT = "Wrong arguments";
+
+    public SortCommand(String argument) {
+        this.argument = argument;
+    }
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> temp_allPersons = addressBook.getAllPersons().immutableListView();
+        if (temp_allPersons.isEmpty()) return new CommandResult(MESSAGE_EMPTY);
+
+        switch (argument) {
+            case " name": {
+                ArrayList<ReadOnlyPerson> allPersons = new ArrayList<>(temp_allPersons);
+
+                Collections.sort(allPersons, new Comparator<ReadOnlyPerson>() {
+                    @Override
+                    public int compare(ReadOnlyPerson o1, ReadOnlyPerson o2) {
+                        return o1.getName().toString().compareToIgnoreCase(o2.getName().toString());
+                    }
+                });
+
+                addressBook.clear();
+
+                for (int i = 0; i < allPersons.size(); i++) {
+                    try {
+                        addressBook.addPerson(new Person(allPersons.get(i)));
+                    }
+                    catch (DuplicateDataException e) {
+                        System.out.println("It wont happen anyway");
+                    }
+                }
+
+                break;
+            }
+            case " phone": {
```

Comment:
In all the switch cases, you should apply the Single Level Abstraction Principle to keep your codes neat and readable.